src/main/java/at/hannibal2/skyhanni/data/HotmData.kt

```diff
@@ -0,0 +1,672 @@
+                HotmAPI.skymall = SkymallPerk.entries.firstOrNull { it.chatPattern.matches(perk) } ?: run {
+                    ErrorManager.logErrorStateWithData(
+                        "Could not read the skymall effect from chat",
+                        "no chatPattern matched",
+                        "chat" to event.message,
+                        "perk" to perk
+                    )
+                    null
+                }
+                ChatUtils.debug("setting skymall to ${HotmAPI.skymall}")
+                return
+            }
+            DelayedRun.runNextTick {
+                mayhemChatPattern.matchMatcher(event.message) {
+                    val perk = group("perk")
+                    HotmAPI.mineshaftMayhem = MayhemPerk.entries.firstOrNull { it.chatPattern.matches(perk) } ?: run {
+                        ErrorManager.logErrorStateWithData(
+                            "Could not read the mayhem effect from chat",
+                            "no chatPattern matched",
+                            "chat" to event.message,
+                            "perk" to perk
+                        )
+                        null
+                    }
+                    ChatUtils.debug("setting mineshaftMayhem to ${HotmAPI.mineshaftMayhem}")
+                }
+            }
+        }
+
+        @SubscribeEvent
+        fun onWorldSwitch(event: IslandChangeEvent) {
+            if (HotmAPI.mineshaftMayhem == null) return
+            HotmAPI.mineshaftMayhem = null
+            ChatUtils.debug("resetting mineshaftMayhem")
+        }
+
+        @SubscribeEvent
+        fun onProfileSwitch(event: ProfileJoinEvent) {
+            HotmAPI.Powder.entries.forEach {
+                if (it.getStorage() == null) {
+                    ProfileStorageData.profileSpecific?.mining?.powder?.put(
+                        it, ProfileSpecificStorage.MiningConfig.PowderStorage()
+                    )
+                }
+            }
+        }
+
+        @SubscribeEvent
+        fun onDebug(event: DebugDataCollectEvent) {
+            event.title("HotM")
+            event.addIrrelevant {
+                add("Tokens : $availableTokens/$tokens")
+                HotmAPI.Powder.entries.forEach {
+                    add("${it.lowName} Powder: ${it.getCurrent()}/${it.getTotal()}")
+                }
+                add("Ability: ${HotmAPI.activeMiningAbility?.printName}")
+                add("Blue Egg: ${HotmAPI.isBlueEggActive}")
+                add("SkyMall: ${HotmAPI.skymall}")
+                add("Mineshaft Mayhem: ${HotmAPI.mineshaftMayhem}")
+            }
+            event.title("HotM - Tree")
+            event.addIrrelevant(
+                entries.filter { it.isUnlocked }.map {
+                    "${if (it.enabled) "✔" else "✖"} ${it.printName}: ${it.activeLevel}"
+                }
+            )
+        }
+    }
+}
+
+private val repoGroup = RepoPattern.group("mining.hotm")
```

Comment:
```suggestion
private val patternGroup = RepoPattern.group("mining.hotm")
```
For consistancy with the rest of the mod